backend/Server/Server.java

```diff
@@ -0,0 +1,279 @@
+            final String rawRequestPath = con.getRequestURI().getRawPath();
+            // Getting file
+            final File file = getFile(rawRequestPath);
+            if (file != null) {
+                // Determine filetype
+                final String filename = file.getName();
+                final Headers headers = con.getResponseHeaders();
+                setReponseContentTypeHeaders(headers, filename);
+                // Sending file bytestream
+                con.sendResponseHeaders(STATUS_CODES.get("OK"), file.length());
+                try (FileInputStream fs = new FileInputStream(file)) {
+                    final byte[] buffer = new byte[0x10000];
+                    int count = 0;
+                    while ((count = fs.read(buffer)) >= 0) {
+                        responseBodyStream.write(buffer, 0, count);
+                    }
+                }
+            } else {
+                // TODO: handle file not found differently based on the type of file requested -Kyle
+                sendTextResponse(con, "File: " + rawRequestPath + " not found.", STATUS_CODES.get("NOT_FOUND"));
+            }
+        }
+    }
+    
+    // Helper function, sends text response back
+    private static void sendTextResponse(final HttpExchange con, final String responseBody, final int statusCode) throws IOException {
+        con.getResponseHeaders().set(HEADER_KEYS.get("CONTENT_TYPE"), HEADER_TYPES.get("TEXT"));
+        con.sendResponseHeaders(statusCode, responseBody.length());
+        con.getResponseBody().write(responseBody.getBytes(CHARSET));
+    }
+    
+    // Helper function for debugging
+    private static void sendDebugResonse(final HttpExchange con, final Map<String, String> paramenters) throws IOException {
+        // Gather request components
+        final URI requestUri = con.getRequestURI();
+        final String requestFragment = getRequestFragment(requestUri.getRawFragment()); // currently broken- uri does not contain fragment for some reason -Kyle
+        final String rawRequestPath = requestUri.getRawPath();
+        
+        final String responseBody = String.format(
+            "{\"uri\": %s, "
+            + "\"params\": %s, "
+            + "\"fragment\": %s, "  
+            + "\"path\": %s}",
+            requestUri,
+            convertToShallowJsonString(paramenters),
+            requestFragment,
+            rawRequestPath
+        );
+        con.getResponseHeaders().set(HEADER_KEYS.get("CONTENT_TYPE"), HEADER_TYPES.get("JSON"));
+        con.sendResponseHeaders(STATUS_CODES.get("OK"), responseBody.length());
+        con.getResponseBody().write(responseBody.getBytes(CHARSET));
+    }
+    
+    private static class ServerConnectionHandler implements HttpHandler {
+        @Override
+        public void handle(final HttpExchange con) throws IOException {
+            int responseStatus = STATUS_CODES.get("NOT_ALLOWED");
+            int responseLength = NO_RESPONSE_LENGTH;

```

Comment:
Unused variables?

Reply:
Yes, forgot to remove them. Thanks for catching that